libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/route/NavigationRouteEx.kt

```diff
@@ -119,6 +119,13 @@ fun NavigationRoute.updateDirectionsRouteOnly(
     return copy(directionsResponse = refreshedResponse)
 }
 
+/**
+ * Used to rebuild any [NavigationRoute] fields that are backed by a native peer, which might've been refreshed.
+ *
+ * At the moment, all fields are `val`s, so a simple re-instantiation is enough.
+ */
+fun NavigationRoute.refreshNativePeer(): NavigationRoute = copy()
```

Comment:
Maybe I'm not getting something but copy also copies the native route, doesn't it? I thought the while point was to refresh native route.

Reply:
The problem is that `NavigationRoute#upcomingRoadObjects` is a field. It's initialized when the object is created (in `NavigationRoute#refresh`) but at that point the native route object is not yet updated. By re-instantiating the class _after_ the native object is update in `MapboxTripSession`, the field is initialized with refreshed data.

There are other ways to do it, I considered using a variable backing field and calling an internal `NavigationRoute#setInternalUpcomingRoadObjects` or something similar. I opted not to do it, because it mutated the instance which could be a surprise to other elements of the codebase that wouldn't be aware of this. By creating a fresh instance, we're not risking weird mutation/timing problems.

Reply:
Ah, OK, thanks for the clarification.